src/test/java/org/openlmis/stockmanagement/service/StockCardAggregateTest.java

```diff
@@ -221,31 +231,42 @@ public void shouldGetStockOutDaysInRange() {
     assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
         LocalDate.of(2018, 5, 12), LocalDate.of(2018, 5, 13)));
 
-    assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
+    assertEquals(new Long(16), stockCardAggregate.getStockoutDays(
         LocalDate.of(2018, 6, 1), LocalDate.of(2018, 6, 30)));
 
-    assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
+    assertEquals(new Long(30), stockCardAggregate.getStockoutDays(
         LocalDate.of(2018, 7, 1), LocalDate.of(2018, 7, 31)));
 
-    assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
+    assertEquals(new Long(30), stockCardAggregate.getStockoutDays(
         LocalDate.of(2018, 8, 1), LocalDate.of(2018, 8, 31)));
 
     assertEquals(new Long(29), stockCardAggregate.getStockoutDays(
         LocalDate.of(2018, 9, 1), LocalDate.of(2018, 9, 30)));
+
+    assertEquals(new Long(0), stockCardAggregate.getStockoutDays(
+            LocalDate.of(2018, 11, 1), LocalDate.of(2018, 11, 30)));
+
+    assertEquals(new Long(16), stockCardAggregate.getStockoutDays(
+            LocalDate.of(2018, 12, 1), LocalDate.of(2018, 12, 31)));
+
+    assertEquals(new Long(30), stockCardAggregate.getStockoutDays(
+            LocalDate.of(2019, 3, 1), LocalDate.of(2019, 3, 31)));
   }
 
   @Test
   public void shouldIncludeStockoutToCurrentDay() {
-    LocalDate date = LocalDate.of(2018, 11, 15);
+    LocalDate date = LocalDate.of(2019, 11, 15);
 
     stockCardAggregate.getCalculatedStockOnHands().add(new CalculatedStockOnHandDataBuilder()
         .withStockCard(stockCard1)
         .withOccurredDate(date)
         .withStockOnHand(0)
         .build());
 
+    long stockoutDaysBeforeDate = stockCardAggregate.getStockoutDays(null, date.minusDays(1));
+
     assertEquals(
-        new Long(1 + DAYS.between(date, LocalDate.now())),
```

Comment:
you should not use the same method in expected and actual. Moreover, I do not understand why you change this expected value

Reply:
I changed it to a hardcoded value

Reply:
ok, looks better. Maybe comment a bit:

// calculated stockout days from first stockCard 2018-05-10 to the last one 2019-11-15